File: voter.react/src/api/models/PollDTO.ts
import type { PollOptionDTO } from './PollOptionDTO';

export interface PollDTO {
  id: number;
  question: string;
  startDate: string;
  endDate: string;
  creatorId: string;
  creatorEmail: string;
  options: PollOptionDTO[];
  creationDate: string;
  isActive: boolean;
  userHasVoted: boolean;
}

File: Voter.Shared/Models/PollDTO.cs

namespace Voter.Shared.Models
{
    public class PollDTO
    {
        public int Id { get; set; }
        public string Question { get; set; } = string.Empty;
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public string CreatorId { get; set; } = string.Empty;
        public string CreatorEmail { get; set; } = string.Empty;
        public List<PollOptionDTO> Options { get; set; } = [];
        public bool UserHasVoted { get; set; }
    }
}
